src/printer.js

```diff
@@ -2900,14 +2937,19 @@ function printArgumentsList(path, options, print) {
     ]);
   }
 
-  const args = path.getValue().arguments;
-
   // This is just an optimization; I think we could return the
   // conditional group for all function calls, but it's more expensive
   // so only do it for specific forms.
   const shouldGroupFirst = shouldGroupFirstArg(args);
   const shouldGroupLast = shouldGroupLastArg(args);
-  if (shouldGroupFirst || shouldGroupLast) {
+  if (
+    !shouldHaveEmptyLineAfterOpeningParen &&
+    !shouldHaveEmptyLineBeforeClosingParen &&
+    hasEmptyLineAfterArgs.every(
+      hasEmptyLineAfterArg => !hasEmptyLineAfterArg
+    ) &&
+    (shouldGroupFirst || shouldGroupLast)
+  ) {
     const shouldBreak = shouldGroupFirst
```

Comment:
So you're not going to preserve empty lines if the first or last argument is expanded?